Ciac 4707/ LOLBas new feed

AradCarmi opened the pull request:
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-4707)

## Description
New feed for LOLBAS API.

dantavori reviewed (changes requested):
good job, see my notes

AradCarmi commented on Packs/FeedLOLBAS/Author_image.png:
The image parsed well on XSOAR, should I still try to fix it?

dantavori replied:
please try fixing, lmk if it keeps showing this message.

AradCarmi replied:
I got confused with the integration logo, do we need this if it is XSOAR supported?

AradCarmi commented on Packs/FeedLOLBAS/Integrations/FeedLOLBAS/FeedLOLBAS.py:
```diff
@@ -0,0 +1,245 @@
+
+''' CONSTANTS '''
+
+BASE_URL = 'https://lolbas-project.github.io/api'
```
Should I expose the base_url to the user or just hide it?

dantavori replied:
yes, put it as a default value of a base_url param. then if the base URL is changed for some reason, users can handle this themselves instead of fixing in the code.

AradCarmi commented on Packs/FeedLOLBAS/Integrations/FeedLOLBAS/FeedLOLBAS.py:
```diff
@@ -0,0 +1,245 @@
+        relationships = create_relationship_list(indicators)
+        if relationships:
+            dummy_indicator_for_relations = {
+                'value': '$$DummyIndicator$$',
```
Not sure there is a best practice for this one, I just followed `FeedMitreAttackv2`.

dantavori replied:
see step 3 here: https://xsoar.pan.dev/docs/integrations/feeds#indicator-objects

dantavori replied:
this is still relevant

AradCarmi commented on Packs/FeedLOLBAS/Integrations/FeedLOLBAS/FeedLOLBAS.py:
```diff
@@ -0,0 +1,245 @@
+                try:
+                    demisto.createIndicators(iter_)
+                except Exception:
+                    # find problematic indicator
```
Again, saw it on FeedMitreAttackV2, it will print the defective indicator for better debugging.

Ni-Knight commented on Packs/FeedLOLBAS/pack_metadata.json:
```diff
@@ -0,0 +1,23 @@
+{
+    "name": "LOLBAS Feed",
+    "description": "Fetches indicators from LOLBAS.",
```
Please have Arik provide a better description, or take the one from the integration.

Ni-Knight commented on Packs/FeedLOLBAS/Integrations/FeedLOLBAS/README.md:
```diff
@@ -0,0 +1,144 @@
+Living off the land binaries" is a term used to describe malware or hacking techniques that take advantage of legitimate tools and processes that are already present on a computer or network, rather than introducing new malware or malicious code. The goal is to blend in with normal activity and avoid detection. Examples of this include using built-in Windows commands to move laterally through a n ...
```
```suggestion
"Living off the land binaries" is a term used to describe malware or hacking techniques that take advantage of legitimate tools and processes that are already present on a computer or network, rather than introducing new malware or malicious code. The goal is to blend in with normal activity and avoid detection. Examples of this include using built-in Windows commands to move laterally through a network, or using scripting languages that are commonly installed on a system to execute malicious code. LOLBAS project is documenting binaries, scripts, and libraries that can be used for Living Off The Land techniques.
```

Ni-Knight commented:
@AradCarmi Please change new fields filenames prefix to `indicatorfield-` and not `incidentfield-`.

Ni-Knight commented on Packs/CommonTypes/IndicatorFields/incidentfield-detections.json:
```diff
@@ -0,0 +1,59 @@
+        {
+            "key": "content",
+            "displayName": "Content",
+            "type": "shortText",
```
```suggestion
            "type": "url",
```

AradCarmi replied:
> @AradCarmi Please change new fields filenames prefix to `indicatorfield-` and not `incidentfield-`.
This is the standard in the CommonType pack.

Ni-Knight commented on Packs/CommonTypes/IndicatorFields/incidentfield-detections.json:
```diff
@@ -0,0 +1,59 @@
+        {
+            "key": "content",
+            "displayName": "Content",
+            "type": "url",
```
```suggestion
            "type": "shortText",
```

My bad, changing back, sorry :(

Ni-Knight commented on Packs/CommonTypes/IndicatorFields/incidentfield-detections.json:
```diff
@@ -0,0 +1,59 @@
+            "key": "content",
+            "displayName": "Content",
+            "type": "url",
+            "orgType": "url",
```
```suggestion
            "orgType": "shortText",
```

Ni-Knight replied:
I was wrong, since some are IOC it doesnt work well

Ni-Knight commented:
@AradCarmi The new fields filenames are still wrong, should be prefix to `indicatorfield-` and not `incidentfield-`.

dantavori commented on Packs/FeedLOLBAS/Integrations/FeedLOLBAS/FeedLOLBAS.py:
```diff
@@ -0,0 +1,300 @@
+            "http": None,
+            "https": None,
+        }
+    return requests.get(MITRE_URL, headers=headers, verify=client.verify, proxies=proxies).json().get('objects', [])
```
please use `BaseClient`'s `_http_request()` with `full_url` arg

dantavori commented on Packs/FeedLOLBAS/Integrations/FeedLOLBAS/FeedLOLBAS.py:
```diff
@@ -0,0 +1,300 @@
+    return mitre_tags
+
+
+def create_indicators(client: Client, pre_indicators) -> List[Dict[str, Any]]:
+    """
+    Create indicators from the response.
```
```suggestion
def build_indicators(client: Client, raw_indicators: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
    """
    Builds indicators JSON data in XSOAR expected format from the raw response.
```

dantavori commented on Packs/FeedLOLBAS/Integrations/FeedLOLBAS/FeedLOLBAS.py:
```diff
@@ -0,0 +1,300 @@
+    return result_map
+
+
+def pre_process_indicator(pre_indicator: Dict[str, Any], mitre_id_to_name) -> List[str]:
+    """
+    Pre-process the indicator, map the MitreID with MitreName and build the relevant tag list.
```
```suggestion
def build_mitre_tags(raw_indicator: Dict[str, Any], mitre_id_to_name: Dict[str, str]) -> List[str]:
    """
    Returns an extended MITRE tags list of a single indicator.
```

dantavori commented on Packs/FeedLOLBAS/Integrations/FeedLOLBAS/FeedLOLBAS.py:
```diff
@@ -0,0 +1,300 @@
+    return requests.get(MITRE_URL, headers=headers, verify=client.verify, proxies=proxies).json().get('objects', [])
+
+
+def map_indicator_fields(pre_indicator: Dict[str, Any]) -> Dict[str, Any]:
```
```suggestion
def build_indicator_custom_fields(pre_indicator: Dict[str, Any]) -> Dict[str, Any]:
```

dantavori reviewed (changes requested):
great progress, we are almost there 🙏🏽

dantavori commented on Packs/FeedLOLBAS/TestPlaybooks/FeedLOLBas_test.yml:
still relevant ☝🏽

ShirleyDenkberg commented on Packs/FeedLOLBAS/Integrations/FeedLOLBAS/FeedLOLBAS.yml:
```diff
@@ -0,0 +1,183 @@
+  required: false
+  type: 8
+  section: Collect
+- additionalinfo: Indicators from this integration instance will be marked with this reputation
```
```suggestion
- additionalinfo: Indicators from this integration instance will be marked with this reputation.
```

ShirleyDenkberg commented on Packs/FeedLOLBAS/Integrations/FeedLOLBAS/FeedLOLBAS_description.md:
```diff
@@ -0,0 +1,5 @@
+## LOLBAS
+#### Living Off The Land Binaries, Scripts and Libraries
+_____
+Import Living of the land binaries from [LOLBAS](https://lolbas-project.github.io/) into the Threat Intelligence Management(TIM) module for security investigations and threat hunting activities.
```
```suggestion
Import Living off the land binaries from [LOLBAS](https://lolbas-project.github.io/) into the Threat Intelligence Management(TIM) module for security investigations and threat hunting activities.
```

ShirleyDenkberg commented on Packs/FeedLOLBAS/Integrations/FeedLOLBAS/README.md:
```diff
@@ -0,0 +1,144 @@
+    | --- | --- | --- |
+    | Base URL |  | True |
+    | Fetch indicators |  | False |
+    | Indicator Reputation | Indicators from this integration instance will be marked with this reputation | False |
```
```suggestion
    | Indicator Reputation | Indicators from this integration instance will be marked with this reputation. | False |
```

ShirleyDenkberg commented:
@Ni-Knight @dantavori @michalgold @idovandijk Doc review is completed.
@AradCarmi Please create a pack readme.

AradCarmi commented on Packs/FeedLOLBAS/Integrations/FeedLOLBAS/README.md:
```diff
@@ -0,0 +1,153 @@
+code. The goal is to blend in with normal activity and avoid detection. Examples of this include using built-in Windows
+commands to move laterally through a network, or using scripting languages that are commonly installed on a system to
+execute malicious code. LOLBAS project is documenting binaries, scripts, and libraries that can be used for Living Off
+The Land techniques.
```
@ShirleyDenkberg can you please have a look also?

ShirleyDenkberg replied:
I also believe that this belongs in the pack readme.  Usually the description file is for providing additional information for setting up the integration.

AradCarmi commented:
Force merge was needed, the build fail on **validate** (IN122) because we updated the TLP selection colors and added the advanced section to feed fields.
This issue can be fixed in the [PR](https://github.com/demisto/demisto-sdk/pull/3003).

dantavori reviewed (approved):
great job - approving but please resolve the one note above
